internal/pkg/service/common/etcdop/op/txn_result.go

```diff
@@ -2,74 +2,38 @@ package op
 
 // TxnResult is result of the TxnOp.
 type TxnResult[R any] struct {
-	result     *Result[R]
-	succeeded  bool
-	subResults []any
+	*resultBase
+	result *R
 }
 
-func newTxnResult[R any](response *RawResponse, result *R) *TxnResult[R] {
-	return &TxnResult[R]{
-		result: newResult[R](response, result),
-	}
+func newTxnResult[R any](base *resultBase, result *R) *TxnResult[R] {
+	return &TxnResult[R]{resultBase: base, result: result}
 }
 
 func newErrorTxnResult[R any](err error) *TxnResult[R] {
-	return newTxnResult[R](nil, nil).AddErr(err)
+	r := newTxnResult[R](newResultBase(nil), nil)
+	r.AddErr(err)
+	return r
 }
 
 func (v *TxnResult[R]) Succeeded() bool {
-	return v.succeeded
-}
-
-func (v *TxnResult[R]) SetSubResults(results []any) *TxnResult[R] {
-	v.subResults = results
-	return v
-}
-
-func (v *TxnResult[R]) SubResults() []any {
-	return v.subResults
-}
-
-func (v *TxnResult[R]) AddSubResult(result any) *TxnResult[R] {
-	v.subResults = append(v.subResults, result)
-	return v
-}
-
-func (v *TxnResult[R]) Response() *RawResponse {
-	return v.result.Response()
-}
```

Comment:
There was option to get transaction sub-results, for each included sub operation/transaction.

It was used minimally. It is possible to define an `OnResut/OnSucceeded'` callback for each sub-operation, so I removed the sub-results. Now, there is only one way to catch sub-results - via callbacks.